### PR TITLE
refactor(secondary-button): remove `linkTo` from component

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- remove `linkTo` prop

reference #1737

### CODE REVIEW

**Review by tdeekens — COMMENTED** [review]
I like it :). Do we now how to approach the `SecondaryButton`s we have internally which act as links?

**Comment by adnasa** [comment]
@tdeekens 
hey, thanks for raising this.
do you mean refactor/migration wise or UX wise?

**Comment by tdeekens** [comment]
> hey, thanks for raising this.
> do you mean refactor/migration wise or UX wise?

More UX wise. Do we plan to continue to have `SecondaryButtons` which act as links and target urls or will we replace them with actual `Link`s? If we ought to keep `SecondaryButton`s which act as links we add an `onClick` with a `router.push` and call it a day - so that's "easy". Still, I was more wondering if this brings a shift in usage for those `SecondaryButtons` with a `linkTo` not acting as a link anymore. 

We can also take this discussion internally elsewhere of course. Was just wondering if there is a UX shift in mindset here already.

**Comment by adnasa** [comment]
That is good consideration.
tbh I have not taken that into consideration at all. I assumed a migration has occurred and had a tunnel view in "just get rid of this, it's been deprecated since a while".

Thanks for raising this. In hindsight, there hasn't been discussion about the migration plan of this as far as I'm aware.
I suggest the following:

1. escalate/initiate internally
2. given that there is a migration plan; we go for this
3. given not, we discard this out of v12.

in the meantime, I'd put this on hold.

**Comment by adnasa** [comment]
@tdeekens please review the following options, or update if I misunderstood you.

### migration options

- replace secondary buttons with links
- add `onClick` for secondary buttons to act as links

**Comment by adnasa** [comment]
@tdeekens 
got rid of `linkTo` but retained `to`..
so we can `<SecondaryButton as={} to={} />`

**Review by tdeekens — APPROVED** [review]
That's nice and more consistent!

**Review by mohib0306 — APPROVED** [review]

**Review by emmenko — COMMENTED** [review]
You have to fix the tests

File: packages/components/buttons/secondary-button/src/secondary-button.js [hunk 1]
```diff
@@ -148,51 +132,16 @@ SecondaryButton.propTypes = {
   },
 
   onClick: requiredIf(PropTypes.func, (props) => {
-    return !props.linkTo && !props.as;
+    return !props.to && !props.as;
   }),
```
Comment (adnasa) [inline packages/components/buttons/secondary-button/src/secondary-button.js:136]: onClick is now required if we don't supply `to` or `as`

**Comment by adnasa** [comment]
oh yeah. tests passed now.
feel free to merge @emmenko when approving

**Comment by adnasa** [comment]
pushing removal of `linkTo` from `Tag`

**Comment by adnasa** [comment]
All righty. `linkTo` is now gone from `<Tag />` as well.

**Comment by emmenko** [comment]
I'm confused now. What's the difference between `linkTo` and `to`? I thought we wanted to remove it 🤔

**Comment by adnasa** [comment]
We remove `linkTo` in favor of `to`

**Comment by mariabarrena** [comment]
> More UX wise. Do we plan to continue to have `SecondaryButtons` which act as links and target urls or will we replace them with actual `Link`s? If we ought to keep `SecondaryButton`s which act as links we add an `onClick` with a `router.push` and call it a day - so that's "easy". Still, I was more wondering if this brings a shift in usage for those `SecondaryButtons` with a `linkTo` not acting as a link anymore.

I've created an investigation ticket on jira to followup on this when we have the time for that (Cannot give a timeline for it yet). For now, there is no UX decision as understanding the root of this request affecting SecondaryButtons only is needed in the team in order to take a proper decision. In any case, after asking in the UX/UI team seems like nobody has ever requested to have SecondaryButtons as links only. I'm only familiar with the FlatButtons behaving like links issue though.

**Comment by adnasa** [comment]
@mariabarrena Okay, thanks for coming back to us on this.
At least we can establish that this is work for follow up

**Comment by adnasa** [comment]
The percy failed snapshot, is due to `hasUnderline` which was removed.
I'll approve the snapshots

**Comment by adnasa** [comment]
oh dear, I merged into `main`... I revert it here https://github.com/commercetools/ui-kit/pull/1833

**Pull request merged**